### PR TITLE
Change assertion in IPGlobalProperties_DomainName_ReturnsEmptyStringWhenNotSet

### PR DESCRIPTION
**Pull request by antonfirsov**
Contributes to #109280. Given that we don't seem to understand the reason the value is `localdomain`, this change feels like a test hack instead of an actual fix, thus the `ActiveIssue` link. If we don't care that much, I'm happy to delete the comment and close the issue.

### CODE REVIEW

**Review by simonrozsival — APPROVED** [review]

File: src/libraries/System.Net.NetworkInformation/tests/FunctionalTests/IPGlobalPropertiesTest.cs [hunk 1]
```diff
@@ -212,7 +212,10 @@ public async Task GetUnicastAddresses_NotEmpty()
         public void IPGlobalProperties_DomainName_ReturnsEmptyStringWhenNotSet()
         {
             IPGlobalProperties gp = IPGlobalProperties.GetIPGlobalProperties();
-            Assert.Equal(string.Empty, gp.DomainName);
+
+            // [ActiveIssue("https://github.com/dotnet/runtime/issues/109280")]
+            string expectedDomainName = PlatformDetection.IsAndroid ? "localdomain" : string.Empty;
```
Comment (wfurt) [inline src/libraries/System.Net.NetworkInformation/tests/FunctionalTests/IPGlobalPropertiesTest.cs:217]: I'm not sure if we re4ally need the comment but I'm OK  keeping it.

**Review by wfurt — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]

**Comment by antonfirsov** [comment]
Outerloop failures are unrelated.

**Pull request merged**